src/helpers/discord_utils.ts

```diff
@@ -214,59 +214,96 @@ export async function sendOptionsMessage(messageContext: MessageContext, guildPr
+    optionStrings[GameOption.DURATION] = guildPreference.getDuration();
 
     for (const gameOption of Object.keys(optionStrings)) {
         const gameOptionString = optionStrings[gameOption];
         if (!gameOptionString) continue;
         optionStrings[gameOption] = (updatedOption && updatedOption.option) === gameOption ? bold(gameOptionString) : codeLine(gameOptionString);
     }
 
-    const goalMessage = `First one to ${optionStrings[GameOption.GOAL]} points wins.`;
-    const guessTimeoutMessage = ` in less than ${optionStrings[GameOption.TIMER]} seconds`;
-    const shuffleMessage = `Songs will be shuffled in ${optionStrings[GameOption.SHUFFLE_TYPE]} order. `;
+    const PREFIX = process.env.BOT_PREFIX;
 
     if (updatedOption && updatedOption.reset) {
-        footerText = `Looking for information on how to use this command? Check out '${process.env.BOT_PREFIX}help [command]' to learn more`;
+        footerText = `Looking for information on how to use a command? Check out '${PREFIX}help [command]' to learn more`;
     }
 
+    const gameOptionArray = Object.values(GameOption);
+    const optionFields: Array<Eris.EmbedField> = [];
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const allOptionFields: Array<Eris.EmbedField> = gameOptionArray.map((option) => {
+        const fieldOption = {
+            name: `${PREFIX}${italicize(GameOptionCommand.get(option))}`,
+            value: optionStrings[option] || "none",
+            inline: true,
+        };
+
+        if (option === GameOption.GROUPS || option === GameOption.SUBUNIT_PREFERENCE) {
+            if (guildPreference.isGroupsMode()) {
+                optionFields.push(fieldOption);
+            }
+        } else if (option === GameOption.GENDER || option === GameOption.ARTIST_TYPE) {
+            if (!guildPreference.isGroupsMode()) {
+                optionFields.push(fieldOption);
+            }
+        } else if (option === GameOption.EXCLUDE) {
+            if (guildPreference.isExcludesMode() && !guildPreference.isGroupsMode()) {
+                optionFields.push(fieldOption);
+            }
+        } else if (option === GameOption.INCLUDE) {
+            if (guildPreference.isIncludesMode() && !guildPreference.isGroupsMode()) {
+                optionFields.push(fieldOption);
+            }
+        } else if (option === GameOption.TIMER) {
+            if (guessTimeoutMode) {
+                optionFields.push(fieldOption);
+            }
+        } else if (option === GameOption.GOAL) {
+            if (goalMode) {
+                optionFields.push(fieldOption);
+            }
+        } else if (option === GameOption.DURATION) {
+            if (guildPreference.isDurationSet()) {
+                optionFields.push(fieldOption);
+            }
+        } else if (GameOptionCommand.has(option)) {
+            optionFields.push(fieldOption);
+        }
+
+        return fieldOption;
+    });
+
     await sendInfoMessage(messageContext,
```

Comment:
From `gameOptionArray` definition to here, remove what's above.

Once you replace the map with a dictionary, you can do:

```ts
    await sendInfoMessage(messageContext,
        {
            title: updatedOption === null ? "Options" : `${updatedOption.option} ${updatedOption.reset ? "reset" : "updated"}`,
            fields: Object.entries(GameOptionCommand).map((option, command) => {
                return {
                    name: `${PREFIX}${italicize(command)}`,
                    value: option,
                    inline: true,
                };

            })
            footerText: footerText !== null ? footerText : null,
            thumbnailUrl: KmqImages.LISTENING,
        });
}
```